biahub/virtual_stain.py

```diff
@@ -0,0 +1,441 @@
+import os
+import shutil
+import subprocess
+
+from pathlib import Path
+from typing import List
+
+import click
+import numpy as np
+import submitit
+
+from iohub.ngff import open_ome_zarr
+
+from biahub.cli.monitor import monitor_jobs
+from biahub.cli.parsing import (
+    input_position_dirpaths,
+    local,
+    monitor,
+    num_processes,
+    output_dirpath,
+    sbatch_filepath_predict,
+    sbatch_filepath_preprocess,
+    sbatch_to_submitit,
+)
+from biahub.cli.utils import create_empty_hcs_zarr
+
+
+def run_viscy_preprocess(
+    data_path: str,
+    num_workers: int = 32,
+    config_file: str = None,
+    path_viscy_env: Path = None,
+    verbose: bool = False,
+):
+    """
+    Run VisCy preprocess on a single FOV.
+    Parameters
+    ----------
+    data_path : str
+        Path to the FOV data.
+    num_workers : int
+        Number of workers to use.
+    config_file : str
+        Path to the VisCy config file.
+    path_viscy_env : Path
+        Path to the VisCy environment.
+    verbose : bool
+        Whether to print verbose output.
+    """
+    cmd = (
+        "module load anaconda && "
+        f"conda activate {path_viscy_env} && "
```

Comment:
Is there a reason to use a separate environment rather than adding an optional dependency on viscy?